config/v1/types_infrastructure.go

```diff
@@ -457,6 +462,32 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+type ClusterDNSConfig struct {
+	// APIServerDNSConfig contains DNS configuration for the API Server.
+	// +optional
+	APIServerDNSConfig []DNSConfig `json:"apiServerDNSConfig,omitempty"`
+
+	// InternalAPIServerDNSConfig contains DNS configuration for the Internal API Server.
+	// +optional
+	InternalAPIServerDNSConfig []DNSConfig `json:"internalAPIServerDNSConfig,omitempty"`
+
+	// IngressDNSConfig contains DNS configuration for cluster services.
+	// +optional
+	IngressDNSConfig []DNSConfig `json:"ingressDNSConfig,omitempty"`
+}
+
+// DNSConfig contains the LB IP and DNS Record type to configure a DNS entry.
+type DNSConfig struct {
+	// recordType is the DNS record type.
```

Comment:
Are there a subset that we support here? Would we expect an MX for example? Could this be an enum of the ones we support?

Reply:
Yes, it would be useful to make this an enum type. Updating.